src/cli.c

```diff
@@ -0,0 +1,57 @@
+#include "cli.h"
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <getopt.h>
+#include <stdio.h>
+
+#define MAX_HZ (1000000L)
+
+static const char* usage_str = "";
+
+__attribute__((__const__)) config_t* cli_config_default(void) {
+    config_t* conf = malloc(sizeof(config_t));
+
+    conf->debug = false;
+    conf->us_delay = 0;
+    conf->rom_path = "";
+
+    return conf;
+}
+
+void cli_config_destroy(config_t *conf) {
+    free(conf);
+}
+
+int cli_config_handle(config_t* const conf, int argc, char **argv) {
+    int c;
+
+    // parse all options first
+    while ((c = getopt(argc, argv, "d")) != -1) {
+        switch (c) {
+            case 'd':
+                conf->debug = true;
+                break;
+            default:
+                fprintf(stderr, "%s\n", usage_str);
+                return EXIT_FAILURE;
+        }
+    }
+
+    // parse the remaining options
+
+    // calculate delay from hertz input
+    long hertz = strtol(argv[optind++], NULL, 10);
```

Comment:
The result from strtol should be checked for errors (see e.g. [some helpful website with a really long url](https://wiki.sei.cmu.edu/confluence/display/c/ERR34-C.+Detect+errors+when+converting+a+string+to+a+number))